Fix #5631: Fix thumbnail loading for multiple classrooms

theMr17 opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

Fixes #5631

This PR introduces a new `ThumbnailImage` composable that integrates the `LessonThumbnailImageView` within a Compose layout. The changes include:

- Adding the basic structure for `ThumbnailImage` composable.
- Updating various UI components to use the new `ThumbnailImage` composable.
- Adding classroom thumbnail drawables to `LessonThumbnailImageView`.

Note: Initial attempts were made to use the Glide Compose library, but was found to be incompatible with other dependencies https://github.com/oppia/oppia-android/issues/5631#issuecomment-2570261190.

## Screenshots

|Light Mode|Dark Mode|
|--|--|
|![image](https://github.com/user-attachments/assets/84d7447e-cde1-4ef5-b08b-a05dce7c6243)|![image](https://github.com/user-attachments/assets/a37ad536-46be-470d-96b6-e953282bc800)|
|![image](https://github.com/user-attachments/assets/c4ad9439-d071-4b73-8c6b-f927a45e420b)|![image](https://github.com/user-attachments/assets/8e472114-5e97-4327-a13f-46cb31f739d9)|

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

theMr17 committed:
Add thumbnail image basic structure

theMr17 committed:
Convert LessonThumbnailImageView to compose

theMr17 committed:
Integrate LessonThumbnailImageView in ThumbnailImage composable (directly inflate xml view)

theMr17 committed:
Use ThumbnailImage in TopicCard

theMr17 committed:
Use ThumbnailImage in all remaining cards

theMr17 committed:
Add missing parameter

theMr17 committed:
Add classroom thumbnail drawables to LessonThumbnailImageView

theMr17 committed:
Fix recomposition of ThumbnailImage and remove extra background colour

theMr17 commented:
Hey @BenHenning, as discussed in chat, we can use this PR to test again with the production assets. PTAL.

theMr17 committed:
Remove unused imports

theMr17 committed:
Fix static checks

BenHenning commented on app/src/main/java/org/oppia/android/app/classroom/ClassroomListViewModel.kt:
```diff
@@ -57,6 +58,7 @@ class ClassroomListViewModel(
   private val profileManagementController: ProfileManagementController,
   private val topicListController: TopicListController,
   private val classroomController: ClassroomController,
+  @ClassroomHtmlParserEntityType private val classroomEntityType: String,
```
This & the following 2 fields don't need the qualifiers since this isn't an inject constructor.

theMr17 replied:
Removed the qualifiers.

BenHenning commented on app/src/main/java/org/oppia/android/app/classroom/ThumbnailImage.kt:
```diff
@@ -0,0 +1,46 @@
+ * currently relies on a traditional Android View approach due to compatibility issues with the
+ * Glide Compose library.
+ *
+ * @param entityId The unique identifier for the entity associated with the thumbnail.
+ * @param entityType The type of the entity (e.g., classroom, topic, story).
+ * @param lessonThumbnail The [LessonThumbnail] containing metadata required to load the image.
+ * @param modifier The [Modifier] to be applied to the layout, defaulting to [Modifier].
```
```suggestion
 * @param entityId the unique identifier for the entity associated with the thumbnail
 * @param entityType the type of the entity (e.g., classroom, topic, story)
 * @param lessonThumbnail the [LessonThumbnail] containing metadata required to load the image
 * @param modifier the [Modifier] to be applied to the layout, defaulting to [Modifier]
```

theMr17 replied:
Updated the kdoc.

BenHenning reviewed (changes requested):
Thanks @theMr17! Can you please attach screenshots to the main comment showing that this works with multiple classrooms enabled at least for test data locally? I think you may need to check out an earlier version of develop (before Gradle was removed) and merge this in to verify (since thumbnails don't currently load for Bazel builds--which is something we will need to fix eventually).

Otherwise, the changes seem quite sensible! Only had a few comments--PTAL.

theMr17 committed:
Update comment

Co-authored-by: Ben Henning <ben@oppia.org>

theMr17 committed:
Remove redundant qualifiers

theMr17 commented:
Hey @BenHenning, thanks for the review! I've addressed your comments and added screenshots to the PR description, PTAL. I didn't need to check out older commits for the Gradle build; image loading works fine for me with Bazel builds as well.

Regarding the classroom thumbnails, they already have a background. The production assets should have the correct background color set for them to display properly.

BenHenning reviewed (approved):
Thanks @theMr17! I'm actually surprised to hear that the Bazel build actually has working thumbnails. Perhaps it's because the JSON assets are still being loaded.

This otherwise LGTM--really happy to see this get in quickly as it will allow us to test multiple classrooms in alpha. :)

BenHenning commented:
Merging with the latest develop & enabling auto-merge.

BenHenning committed:
Merge branch 'develop' into fix-thumbnails-multiple-classrooms
